_posts/2025-03-26-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,702 @@
+- Strimzi Custom Resource Definitions (CRDs) are only guaranteed to be backwards compatible between adjacent versions. So, while it is possible to have more than two versions installed, you should aim to only have two adjacent Strimzi versions running at any one time.
+- Changes in CRD API versions (`v1beta2` to `v1` for example) require all Kafka clusters to migrate to the new API version before a newer version of Strimzi is installed. 
+- Special care needs to be taken when removing old Strimzi versions. The CRDs should be handled separately and great care should be taken when upgrading them. Removing the CRDs (even temporarily), rather than upgrading them, would cause all Strimzi managed Kafka clusters (and other operands) to be deleted.
+
+### Contents
+
+- [Background](#background)
+   - [Custom Resource Definitions](#custom-resource-definitions)
+   - [Strimzi Deployment](#strimzi-deployment)
+- [Options for phased upgrades](#options-for-phased-upgrades)
+   - [Strimzi per Namespace](#strimzi-per-namespace)
+   - [Pause Reconciliation](#pause-reconciliation)
+   - [Resource Selector](#resource-selector)
+- [Resource Selector based upgrade](#resource-selector-based-upgrade)
+   - [Concurrent Strimzi Deployments](#concurrent-strimzi-deployments)
+      - [Custom Resource Definitions](#custom-resource-definitions-1)
+      - [Strimzi install resources](#strimzi-install-resources)
+      - [Deploying Strimzi with Kustomize](#deploying-strimzi-with-kustomize)
+      - [Renaming Resources](#renaming-resources)
+      - [Patching Environment Variables](#patching-environment-variables)
+      - [Patching Role and ClusterRole Bindings](#patching-role-and-clusterrole-bindings)
+      - [Watching all namespaces](#watching-all-namespaces)
+      - [Operator Set](#operator-set)
+   - [The Phased Upgrade Process](#the-phased-upgrade-process)
+   - [Things to watch out for](#things-to-watch-out-for)
+   - [Further deployment options](#further-deployment-options)
+- [Conclusion](#conclusion)
+
+### Background
+
+Strimzi handles the [upgrade of Kafka clusters](https://strimzi.io/docs/operators/latest/deploying#proc-upgrade-kafka-kraft-str) for you, making it simple and easy for infrastructure teams to handle these complex operations. 
+The [upgrade of the Strimzi operator](https://strimzi.io/docs/operators/latest/deploying#assembly-upgrade-cluster-operator-str) itself is also straight-forward. 
+However, when you have a large fleet of Kafka clusters being managed by Strimzi, there are several elements you must keep in mind when performing the upgrade. 
+
+#### Custom Resource Definitions 
+
+The first step in a Strimzi operator upgrade is to update the [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) (CRDs) for the Strimzi Custom Resources (CRs): `Kafka`, `KafkaTopic`, `KafkaUser` etc (we refer to these as the Strimzi operator's _operands_). 
```

Comment:
For me, the operand is the running _component_ operated by Strimzi. Not sure I would call the CR _operand_ - even double so for topics and users which do not have the traditional style _operand_ at all.